doc/api_base.rst

```diff
@@ -52,10 +56,12 @@
   .. method:: add_remote_rpms(path_list, strict=True, progress=None)
 
     Add RPM files at list `path_list` to the :attr:`sack` and return the list of respective
-    :class:`dnf.package.Package` instances. Does the download to a temporary files for each path if
-    `path` is a remote URL. Raises :exc:`IOError` if there are problems obtaining during reading
-    files and `strict=True`. `progress`, if given, should be a :class:`.DownloadProgress` and can be
-    used by the caller to monitor the progress of the download.
+    :class:`dnf.package.Package` instances. It must be called before anything is added into
+    :attr:`goal`. Does the download to a temporary files for each path if `path` is a remote URL.
+    Raises :exc:`IOError` if there are problems obtaining during reading files and `strict=True`.
+    Raises :exc:`dnf.exceptions.Error` if the goal is not empty. `progress`, if given, should be a
+    :class:`.DownloadProgress` and can be used by the caller to monitor the progress of the
+    download.
```

Comment:
The whole paragraph is pretty rough, lets take the opportunity to clean it up. I've also put your sentence as the first one, becase it was interrupting the logical flow of the sentences before and after it. An alternative would be to put it at the end, which I slightly prefer, but I assume you want it to be more visible to the reader.

> **This function** must be called before anything is added **to the** :attr:`goal`. **Adds** RPM files
**in** `path_list` to the :attr:`sack` and return the list of respective :class:`dnf.package.Package`
instances. **Downloads the RPMs** to a temporary **file** for each path if **it** is a remote URL.
Raises :exc:`IOError` if there are problems during **???** files and `strict=True`.
Raises :exc:`dnf.exceptions.Error` if the goal is not empty. `progress`, if given, should be a
:class:`.DownloadProgress` **instance** **which** can be **used to** monitor the progress of the
download.

Note the question marks, I don't know what is the operation that can raise the `IOError`, the current formulation really doesn't make any sense...